Lib/shutil.py

```diff
@@ -234,26 +229,36 @@ def copyfile(src, dst, *, follow_symlinks=True):
             # XXX What about other special files? (sockets, devices...)
             if stat.S_ISFIFO(st.st_mode):
                 raise SpecialFileError("`%s` is a named pipe" % fn)
+            if _WINDOWS and i == 0:
+                file_size = st.st_size
 
     if not follow_symlinks and os.path.islink(src):
         os.symlink(os.readlink(src), dst)
     else:
         with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
+            # Linux / Solaris
             if _HAS_SENDFILE:
                 try:
                     _fastcopy_sendfile(fsrc, fdst)
                     return dst
                 except _GiveupOnFastCopy:
                     pass
-
-            if _HAS_FCOPYFILE:
+            # macOS
+            elif _HAS_FCOPYFILE:
                 try:
-                    _fastcopy_osx(fsrc, fdst, posix._COPYFILE_DATA)
+                    _fastcopy_fcopyfile(fsrc, fdst, posix._COPYFILE_DATA)
                     return dst
                 except _GiveupOnFastCopy:
                     pass
+            # Windows: for files >= 128 MiB in size we observe a
+            # considerable speedup by using a readinto()/memoryview()
+            # variant of copyfileobj(), see:
+            # https://github.com/python/cpython/pull/7160#discussion_r195162475
+            elif _WINDOWS and file_size >= 128 * 1024 * 1024:
```

Comment:
Would you mind to add a top-level private constant rather than an hardcoded value? (Would it make sense to make it somehow public? Or add an optional parameter...?)

Reply:
This is outdated, I modified code to always use the readinto() variant regardless of file size, see: https://github.com/python/cpython/pull/7160#discussion_r195458353